.pre-commit-config.yaml

```diff
@@ -41,8 +41,8 @@ repos:
     hooks:
       - id: validate-configs-syntax
         name: validate-configs-syntax
-        entry: python3 ./snuba/validate_configs.py
-        language: system
-        pass_filenames: false
+        entry: python3 -m snuba.validate_configs
+        language: python
+        additional_dependencies: [ 'jsonschema', 'pyyaml' ]
```

Comment:
this should also set a `files` filter such that it's running on specific filenames -- and the validator itself should take specific filenames and only validate those files

Reply:
Currently, the config files are recursed through a file path specified in a settings variable (https://github.com/getsentry/snuba/blob/5786877b22c8ce24b9cbc75142ace77fdf6a5410/snuba/validate_configs.py#L14).  Also, this validator is used in our CI via `make` command (https://github.com/getsentry/snuba/blob/347aab5c3f117cd3ab4114f347386037941f7290/.github/workflows/ci.yml#L100). Passing config filenames will require a different validation function, and the additional maintenance of where configs live. Would it be possible to leave this?

Reply:
as it is right now it's going to always validate every config every time even when it's not changed (which on my machine adds about half a second when committing (even when committing unrelated things!)) -- maybe that overhead is ok but it's right on the borderline where I'd consider turning it off

Reply:
That's a good idea, our other pre-commit hooks are only triggered when the relevant files are changed.

Reply:
Ah I misunderstood the purpose of the `files` flag. Updated the pre-commit hook to only execute if files within `snuba/datasets/configuration` has been modified (includes configs, loaders, jsonschemas, etc.)

Reply:
a further improvement would be to make the hook only check the files it is passed rather than all of them whenever triggered

as it is right now you probably need `pass_filenames: false` otherwise it's going to try and run it in parallel (`pass_filenames` is usually a hack, the right thing to do is to make it respond to just the filenames passed to it)